apps/frontend/src/libs/hooks/use-app-search/use-app-search.hook.ts

```diff
@@ -0,0 +1,34 @@
+import { useCallback } from "react";
+
+import { useAppForm } from "~/libs/hooks/hooks.js";
+
+type SearchFormValues = {
+	[key: string]: string;
+};
+
+type UseSearchResult = {
+	control: ReturnType<typeof useAppForm<SearchFormValues>>["control"];
+	errors: ReturnType<typeof useAppForm<SearchFormValues>>["errors"];
+	handleSearch: () => void;
+	name: keyof SearchFormValues;
+};
+
+const useAppSearch = (
+	callback: (value: string) => void,
+	fieldName: string = "searchField",
+): UseSearchResult => {
+	const { control, errors, getValues } = useAppForm<SearchFormValues>({
```

Comment:
This hook shouldn't return control, errors and other things. It should receive the callback, parameter name and handle all logic